### DISCUSSION TITLE
Debugging VS Code Extensions Through SSH is Extremely Slow

### DISCUSSION THREAD

**Discussion by RodogInfinite** [coder/code-server#4906] [General] [answered]
I'm just getting into [extension creation](https://code.visualstudio.com/api/get-started/your-first-extension) and have found that when I'm developing through SSH and I try to launch or refresh the Extension Development Host (EDH) window tab it takes about 25 seconds to get the new tab to populate. After that it takes about another 15 seconds to display the "Hello World!" information message. This is in contrast to nearly instantaneous opening of the EDH tab and displaying of the message when I'm working directly on the host machine. I'm using `ssh -L 8080:127.0.0.1:8080 [user]@<instance-ip>` to get into the machine, then I run my Singularity container which launches `code-server` that immediately becomes available in the browser. 

I can switch to developing the extension directly on the host machine, but this will severely limit development speed and makes me worry about if the same underlying issue will impact the ability for data to be presented through the extension after it's completed.

Any suggestions for fixing this would be extremely helpful!

**Comment by jsjoeio** [comment]
@code-asher any ideas on why extension development over SSH would be so slow? 🤔

**Comment by RodogInfinite** [comment, marked as answer]
Looks like it was just a bunch of zombie processes gumming up the works. No direct causes I could find, but solved with restarting everything after an instance was open for a long time.
